src/main/java/com/targus/ui/controllers/InformativeController.java

```diff
@@ -0,0 +1,77 @@
+package com.targus.ui.controllers;
+
+import com.targus.base.Solution;
+import com.targus.problem.wsn.WSN;
+import com.targus.problem.wsn.WSNMinimumSensorObjective;
+import com.targus.problem.wsn.WSNOptimizationProblem;
+import com.targus.represent.BitString;
+import com.targus.ui.Mediator;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextField;
+
+import java.util.HashSet;
+
+public class InformativeController {
+    @FXML
+    TextField txtSensorObjective;
+    @FXML
+    TextField txtWeightSensorObjective;
+    @FXML
+    TextField txtWeightSensorObjectiveResult;
+    @FXML
+    TextField txtConnectivityObjective;
+    @FXML
+    TextField txtWeightConnectivityObjective;
+    @FXML
+    TextField txtWeightConnectivityObjectiveResult;
+    @FXML
+    TextField txtCoverageObjective;
+    @FXML
+    TextField txtWeightCoverageObjective;
+    @FXML
+    TextField txtWeightCoverageObjectiveResult;
+    @FXML
+    TextField txtTotalResult;
+
+    private Mediator mediator;
+
+    public void setMediator(Mediator mediator) {
+        this.mediator = mediator;
+    }
+
+    public void display() {
+        WSNOptimizationProblem wsnOptimizationProblem = (WSNOptimizationProblem) mediator.getOptimizationProblem();
+        Solution solution = mediator.getSolution();
+
+        if (wsnOptimizationProblem == null || solution == null) {
+            return;
```

Comment:
The justification for that if condition is that display method is invoked pretty much after each action that would cause a change in the optimization problem object such as adding a new sensor or a target. In such cases, there is the possibility that display function will be called even though there are uninitialized variables for displaying. This solution was an answer to that.